explainers/signed-exchange-alternate-link-explainer.md

```diff
@@ -0,0 +1,164 @@
+
+### Recursive subresource signed exchange prefetch
+1. A crawler of an aggregator site fetches an article (https://publisher.example/article.html).
+1. There is an alternate link of signed exchange of the article in the response header:
+    ```
+    Link: <https://sxg.publisher.example/article.html.sxg>;
+            rel="alternate";
+            type="application/signed-exchange;v=b3";
+            anchor="https://publisher.example/article.html"
+    ```
+1. The crawler fetches and verifies the signed exchange (article.html.sxg).
+
+   1. The response from the server has an alternate link of subresource signed exchange:
+      ```
+      Link: <https://sxg.publisher.example/lib.js.sxg>;
+              rel="alternate";
+              type="application/signed-exchange;v=b3";
+              anchor="https://cdn.publisher.example/lib.js"
+      ```
+   1. And the inner response of the main resource signed exchange (article.html.sxg) has a preload header and an [allowed-alt-sxg header](./signed-exchange-subresource-subtitution-explainer.md):
+      ```
+      Link: <https://cdn.publisher.example/lib.js>;
+              rel="preload";
+              as="script"
+      Link: <https://cdn.publisher.example/lib.js>;
+              rel="allowed-alt-sxg";
+              header-integrity="sha256-XXXXXX"
+      ```
+1. The crawler fetches and verifies the subresource signed exchange (lib.js.sxg).
+1. The aggregator site will serve the signed exchanges (article.html.sxg and lib.js.sxg) from their own server.
+
+1. While a user is browsing the aggregator site (https://feed.example), a prefetch link element is inserted because the use is likely to want to read the article.
+   ```
+   <link rel="prefetch" href="https://feed.example/sxg.publisher.example/article.html.sxg">
+   ```
+1. The UA prefetches the signed exchange (article.html.sxg).
+   1. The response from the server has an alternate link of subresource signed exchange:
+      ```
+      Link: <https://feed.example/sxg.publisher.example/lib.js.sxg>;
+              rel="alternate";
+              type="application/signed-exchange;v=b3";
+              anchor="https://cdn.publisher.example/lib.js"
+      ```
+   1. And the inner response of the main resource signed exchange (article.html.sxg) has a preload header and an allowed-alt-sxg header:
+      ```
+      Link: <https://cdn.publisher.example/lib.js>;
+              rel="preload";
+              as="script"
+      Link: <https://cdn.publisher.example/lib.js>;
+              rel="allowed-alt-sxg";
+              header-integrity="sha256-XXXXXX"
+      ```
+1. The UA recursively prefetches lib.js.sxg.
+1. If the user clicks a link the article’s signed exchange, both the main resource of the article and the script are loaded from the prefetched signed exchanges.
+
+## Proposal
+While processing preload link HTTP headers in prefetched main resource signed exchange’s inner response:
```

Comment:
```suggestion
While prefetching an HTML resource:
```

Then write an explicit loop that checks if every preload has a matching and allowed prefetch before deciding to use all of the prefetches.

Reply:
Done